client/src/components/Ticketing/ticketingmanager/Season/components/utils/apiRequest.ts

```diff
@@ -0,0 +1,102 @@
+import {formatSeasonDate} from '../../seasonUtils';
+
+export interface RequestBody {
+  seasonid: number;
+  name: string;
+  startdate: number;
+  enddate: number;
+  imageurl: string;
+}
+
+export const createNewSeason = async (reqBody: RequestBody, token: string) => {
+  try {
+    const createSeasonRes = await fetch(
+      process.env.REACT_APP_API_2_URL + '/season',
+      {
+        credentials: 'include',
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`,
+        },
+        body: JSON.stringify(reqBody),
+      },
+    );
+
+    if (!createSeasonRes.ok) {
+      throw new Error(`Failed to create new season`);
+    }
+
+    const {seasonid} = await createSeasonRes.json();
+    return seasonid;
+  } catch (error) {
+    console.error(error);
+    return null;
+  }
+};
+
+export const getSeasonInfo = async (seasonId: number, token: string) => {
+  if (seasonId !== 0) {
+    try {
+      const getSeasonRes = await fetch(
+        process.env.REACT_APP_API_2_URL + `/season/${seasonId}`,
+        {
+          credentials: 'include',
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            'Authorization': `Bearer ${token}`,
+          },
+        },
+      );
+
+      if (!getSeasonRes.ok) {
+        throw new Error('Failed to fetch season information');
+      }
+
+      // Converting startdate and enddate response to string for form validation
+      const seasonInfo = await getSeasonRes.json();
+      const {startdate: sdate, enddate: edate} = seasonInfo;
+      const modifiedSeasonInfo = {
+        ...seasonInfo,
+        startdate: formatSeasonDate(sdate, true),
+        enddate: formatSeasonDate(edate, true),
+      };
+
+      return modifiedSeasonInfo;
+    } catch (error) {
+      console.error(error);
+      return null;
+    }
+  }
+};
+
+export const updateSeasonInfo = async (
+  reqBody: RequestBody,
+  seasonId: number,
+  token: string,
+) => {
+  try {
+    const updateSeasonRes = await fetch(
+      process.env.REACT_APP_API_2_URL + `/season/${seasonId}`,
+      {
+        credentials: 'include',
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`,
+        },
+        body: JSON.stringify(reqBody),
+      },
+    );
+
+    if (!updateSeasonRes.ok) {
```

Comment:
Is it possible for it not to throw an error, and also not be `ok`?

Reply:
The error is being thrown only if the HTTP status code isn't a success 200 level code, so if I'm understanding your question correctly, no its not possibly because if it is not ok, that means a 400 or 500 level status code was return which means an error has occurred.

Reply:
Ok, well if that’s the case, I don’t think that if clause will ever be reached because it would have already been cause by the `except`

Reply:
Sorry, I didn't explain that very well. If the status code response from the API is 400 or 500 which indicates an error (either b/c of the client or server), Javascript won't throw an error because technically since the API responded with a status code, it is considered a success. Therefore, if we didn't have this if statement to check if the status code was 200 and we got a 400 or 500 level status code, the catch clause won't ever be executed.

Reply:
Oh ok, I see. Makes sense